src/main.cpp

```diff
@@ -17,37 +34,66 @@ int main(int argc, char** argv) {
     setenv("MOZ_ENABLE_WAYLAND", "1", 1);
     setenv("XDG_CURRENT_DESKTOP", "Hyprland", 1);
 
+
     // parse some args
     std::string configPath;
-    bool        ignoreSudo = false;
-    for (int i = 1; i < argc; ++i) {
-        if (!strcmp(argv[i], "--i-am-really-stupid"))
+    bool ignoreSudo = false;
+
+    // Opinion: All of this code is a mess. Try to do something with that, folks? Pretty please?
+    // Some code might be not so good. I'm working at that at 3 am.
+
+    // You can remove that.
+    std::cout << "Argument parsing and code reformatting patch (nitrogenez@github.com)" << std::endl;
+
+    std::vector<std::string> args { argv + 1, argv + argc };
+
+    for (auto it = args.begin(); it != args.end(); it++) {
+        if (it->compare("--i-am-really-stupid") == 0 && !ignoreSudo) {
+            std::cout << "[ WARNING ] Running Hyprland with superuser privileges might damage your system" << std::endl;
+
             ignoreSudo = true;
-        else if ((!strcmp(argv[i], "-c") || !strcmp(argv[i], "--config")) && argc >= i + 2) {
-            configPath = std::string(argv[++i]);
-            Debug::log(LOG, "Using config location %s.", configPath.c_str());
-        } else {
-            std::cout << "Hyprland usage: Hyprland [arg [...]].\n\nArguments:\n"
-                      << "--help         -h | Show this help message\n"
-                      << "--config       -c | Specify config file to use\n";
-            return 1;
         }
-    }
 
-    if (!ignoreSudo) {
-        if (Init::isSudo()) {
-            std::cout << "Hyprland shall not be run as the root user. If you really want to, use the --i-am-really-stupid flag.\n";
-            return 1;
+        else if (it->compare("-c") == 0 || it->compare("--config") == 0) {
+            if (std::next(it)->c_str() == nullptr) {
+                help();
+                return 1;
+            }
+            std::string next_arg = std::next(it)->c_str();
+
+            if (!std::filesystem::exists(next_arg)) {
+                std::cerr << "[ ERROR ] Config path '" << next_arg << "' doesn't exist!" << std::endl;
+                help();
+
+                return 1;
+            }
+
+            configPath = next_arg;
+            Debug::log(LOG, "User-specified config location: '%s'", configPath.c_str());
+            continue;
         }
-    } else {
-        std::cout << "Running with ignored root checks, I surely hope you know what you're doing.\n";
-        sleep(1);
+    }
+
+    // Funny thing.
+    // We don't *really* need that check, because in most cases `root` user
+    // doesn't have XDG_RUNTIME_DIR envar set, which causes runtime error.
+
+    // But if set, then yes, that needs to be checked.
+
+    // We don't need to display that message when no superuser privs are found
+    if (!ignoreSudo && Init::isSudo()) {
+        std::cerr << "[ ERROR ] Hyprland was launched with superuser priveleges, but the privileges check is not omitted." << std::endl;
```

Comment:
You removed the hint for the user to use `--i-am-really-stupid` to omit it

Reply:
If the user needs to omit the check, just see --help and see what arguments are there.

Reply:
you vastly overestimate the intelligence of an average user.